File: 28. Find the Index of the First Occurrence in a String/Solution-for-28.cpp
#include <iostream>
#include <string>

class Solution {
public:
    int strStr(std::string haystack, std::string needle) {
        // If needle is an empty string, return 0
        if (needle.empty()) {
            return 0;
        }
        
        // Find the position of needle in haystack
        size_t pos = haystack.find(needle);
        
        // If found, return the position; otherwise, return -1
        return (pos != std::string::npos) ? static_cast<int>(pos) : -1;
    }
};

// // Example usage
// int main() {
//     Solution solution;

//     // Output the result
//     std::cout << solution.strStr("sadbutsad", "sad") << std::endl; // 0
//     std::cout << solution.strStr("leetcode", "leeto") << std::endl; // -1

//     return 0;
// }

File: 28. Find the Index of the First Occurrence in a String/Solution-for-28.js
/**
 * Finds the index of the first occurrence of a substring (needle) in a string (haystack).
 * Returns -1 if the substring is not found.
 *
 * @param haystack - The string in which to search for the substring.
 * @param needle - The substring to search for.
 * @returns The index of the first occurrence of the substring, or -1 if not found.
 */
function strStr(haystack, needle) {
    return haystack.search(needle);
};

// // Example usage
// // Output the result
// console.log(strStr("sadbutsad", "sad")); // 0
// console.log(strStr("leetcode", "leeto")); // -1
